source/s3_meta_request.c

```diff
@@ -1064,6 +1058,17 @@ static void s_s3_meta_request_request_on_signed(
 
 finish:
 
+    /**
+     * Add the x-amz-content-sha256 header to support trailing checksum.
+     */
+    if (error_code == AWS_ERROR_SUCCESS && request->send_data.require_streaming_unsigned_payload_header) {
```

Comment:
is it possible to move this earlier in the function. It's an odd flow, having it after the `finish` goto label, and with the `if (error_code == AWS_ERROR_SUCCESS`